### PR TITLE
Module1_API

### PR DESCRIPTION
**Pull request by tatsianaikan**

### CODE REVIEW

File: src/main/java/com/example/bookshop/controller/BookingController.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+
+    /**
+     * Create one new booking in DB
+     * @param bookingObj Object
+     * @return a JSON with created booking
+     */
+    @PostMapping("/createBooking")
+    @ResponseStatus(HttpStatus.CREATED)
+    public Booking createBooking(@RequestBody BookingObj bookingObj) {
```
Comment (AndriiFrolov) [inline src/main/java/com/example/bookshop/controller/BookingController.java]: A bit unclear why we have 2 methods to create bookings

Comment (AndriiFrolov) [inline src/main/java/com/example/bookshop/controller/BookingController.java]: seems strange to me that you have BookingObj (better use DTO naming) as input and Booking as output
If you have DTOs, better to communicate with external world using them

Imagine situation. You are frontend developer. You have CreateBooking page. User clicks 'Book button'. As a developer you send BookingDTO to backend, but receive Booking entity in response. On next page your user realizes that he needs to update his booking. ANd as a developer you have to transform your Booking entity to BookingDto (cause your EDIT api takes BookingDto as input)

File: src/main/java/com/example/bookshop/controller/BookingController.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+     * Delete Booking by Id
+     * @param bookingId - int
+     * @return String message about successful removal of Booking
+     */
+    @DeleteMapping("/delete")
+    @ResponseBody
+    public String deleteBooking(@RequestParam int bookingId){
+        Booking booking = new Booking();
+        booking = bookingService.getBooking(bookingId);
+        bookingService.deleteBooking(bookingId);
```
Comment (AndriiFrolov) [inline src/main/java/com/example/bookshop/controller/BookingController.java:80]: whole logic better to move to BookingService itself

File: src/test/java/com/example/bookshop/BookingControllerTests.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    @Autowired
+    private BookingController bookingController;
+    @Autowired
+    private UserController userController;
+
+    UserObj userObj;
+
+
+    @Test
+    public void addNewBooksObj_Status200() {
```
Comment (AndriiFrolov) [inline src/test/java/com/example/bookshop/BookingControllerTests.java]: I don't see status 200 is actually tested here. That could be nice test for Service layer (where some logic happens)
If you test controlller, make sense to actually send API calls and verify status codes, headers, responces
https://www.baeldung.com/spring-boot-testing

**Review by AndriiFrolov — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/bookshop/controller/BookingController.java (hunk 1 above)
Comment (tatsianaikan) [reply src/main/java/com/example/bookshop/controller/BookingController.java]: Just as example for me demonstrates what I can use @RequestParam or @RequestBody in parameters

Comment (tatsianaikan) [reply src/main/java/com/example/bookshop/controller/BookingController.java]: agree with you. Thanks

File: src/main/java/com/example/bookshop/controller/BookingController.java (hunk 2 above)
Comment (tatsianaikan) [reply src/main/java/com/example/bookshop/controller/BookingController.java:80]: you are right! I update it.

File: src/test/java/com/example/bookshop/BookingControllerTests.java (hunk 3 above)
Comment (tatsianaikan) [reply src/test/java/com/example/bookshop/BookingControllerTests.java]: got it!

**Pull request merged**